akka-stream/src/main/scala/akka/stream/impl/ActorMaterializerImpl.scala

```diff
@@ -294,46 +294,12 @@ private[akka] object ActorProcessorFactory {
-    def interpAttr(s: Stage[_, _], newAttributes: Attributes): (Props, Unit) = (ActorInterpreter.props(settings, List(s), materializer, newAttributes), ())
-    def inputSizeAttr(n: Long) = {
-      if (n <= 0)
-        inputBuffer(initial = 1, max = 1) and att
-      else if (n <= materializer.settings.maxInputBufferSize)
-        inputBuffer(initial = n.toInt, max = n.toInt) and att
-      else
-        att
-    }
     op match {
-      case Map(f, _)                  ⇒ interp(fusing.Map(f, settings.supervisionDecider))
-      case Filter(p, _)               ⇒ interp(fusing.Filter(p, settings.supervisionDecider))
-      case Drop(n, _)                 ⇒ interp(fusing.Drop(n))
-      case Take(n, _)                 ⇒ interpAttr(fusing.Take(n), inputSizeAttr(n))
-      case TakeWhile(p, _)            ⇒ interp(fusing.TakeWhile(p, settings.supervisionDecider))
-      case DropWhile(p, _)            ⇒ interp(fusing.DropWhile(p, settings.supervisionDecider))
-      case Collect(pf, _)             ⇒ interp(fusing.Collect(pf, settings.supervisionDecider))
-      case Scan(z, f, _)              ⇒ interp(fusing.Scan(z, f, settings.supervisionDecider))
-      case Fold(z, f, _)              ⇒ interp(fusing.Fold(z, f, settings.supervisionDecider))
-      case Recover(pf, _)             ⇒ interp(fusing.Recover(pf))
-      case Expand(s, f, _)            ⇒ interp(fusing.Expand(s, f))
-      case Conflate(s, f, _)          ⇒ interp(fusing.Conflate(s, f, settings.supervisionDecider))
-      case Buffer(n, s, _)            ⇒ interp(fusing.Buffer(n, s))
-      case MapConcat(f, _)            ⇒ interp(fusing.MapConcat(f, settings.supervisionDecider))
-      case MapAsync(p, f, _)          ⇒ interp(fusing.MapAsync(p, f, settings.supervisionDecider))
-      case MapAsyncUnordered(p, f, _) ⇒ interp(fusing.MapAsyncUnordered(p, f, settings.supervisionDecider))
-      case Grouped(n, _)              ⇒ interp(fusing.Grouped(n))
-      case Sliding(n, step, _)        ⇒ interp(fusing.Sliding(n, step))
-      case Log(n, e, l, _)            ⇒ interp(fusing.Log(n, e, l))
-      case GroupBy(f, _)              ⇒ (GroupByProcessorImpl.props(settings, f), ())
-      case PrefixAndTail(n, _)        ⇒ (PrefixAndTailImpl.props(settings, n), ())
-      case Split(d, _)                ⇒ (SplitWhereProcessorImpl.props(settings, d), ())
-      case ConcatAll(_)               ⇒ (ConcatAllImpl.props(materializer), ())
-      case StageFactory(mkStage, _)   ⇒ interp(mkStage())
-      case MaterializingStageFactory(mkStageAndMat, _) ⇒
-        val s_m = mkStageAndMat()
-        (ActorInterpreter.props(settings, List(s_m._1), materializer, att), s_m._2)
+      case GroupBy(f, _)         ⇒ (GroupByProcessorImpl.props(settings, f), ())
+      case PrefixAndTail(n, _)   ⇒ (PrefixAndTailImpl.props(settings, n), ())
+      case Split(d, _)           ⇒ (SplitWhereProcessorImpl.props(settings, d), ())
+      case ConcatAll(_)          ⇒ (ConcatAllImpl.props(materializer), ())
```

Comment:
(but no, should not go into this PR, let’s do things in a somewhat orderly fashion)